lib/processors/background/BackgroundProcessor.ts

```diff
@@ -242,42 +256,70 @@ export abstract class BackgroundProcessor extends Processor {
       this._webgl2Pipeline = null;
     }
 
-    if (!this._webgl2Pipeline && this._pipeline === Pipeline.WebGL2) {
-      this._createWebGL2Pipeline(inputFrame as HTMLVideoElement, captureWidth, captureHeight, inferenceWidth, inferenceHeight);
+    if (this._pipeline === Pipeline.WebGL2) {
+      if (!this._webgl2Pipeline) {
+        this._createWebGL2Pipeline(
+          inputFrameBuffer as HTMLVideoElement,
+          captureWidth,
+          captureHeight,
+          inferenceWidth,
+          inferenceHeight
+        );
+      }
+      this._webgl2Pipeline?.sampleInputFrame();
```

Comment:
It seems we're creating _webgl2Pipeline only once. How are we updating the widths and heights params?

Reply:
For example, when we are setting a new background, we clean up the pipeline, which triggers this code to recreate it with the new height and width in the next `processFrame()` call.